### PR TITLE
feat: support multiple dimensions from same column

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #3232

### Description:

Documentation PR: https://github.com/lightdash/lightdash/pull/8167

Main changes:
- [x] update seed data
- [x] update dbt types to support `additional_dimensions`
- [x] update yml validation
- [x] create a new dimension for each entry in `additional_dimensions`
- [x] track how many additional dimensions are in each project compile
- [x] add unit test

Example data:
```
id,plan_name,metadata
1,free,{"created_by":"John Miller"}
2,silver,{"created_by":"Elliot Stone"}
3,gold,{"created_by":"John Miller"}
4,platinum,{"created_by":"Elliot Stone"}
5,diamond,{"created_by":"John Miller"}
```
Example yml:
```
version: 2
models:
  - name: plan
    columns:
      - name: id
      - name: plan_name
      - name: metadata
        meta:
          dimension:
            hidden: true
          additional_dimensions:
            created_by:
              type: string
              sql: ${plan.metadata}->>'created_by'
            created_by_first_name:
              type: string
              sql: SPLIT_PART(${created_by}, ' ', 1)
            created_by_last_name:
              type: string
              sql: SPLIT_PART(${created_by}, ' ', 2)
          metrics:
            creator_count:
              type: count
              sql: ${created_by}
```

Screenshot 2023-11-30 at 14 27 12

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Review by TuringLovesDeathMetal — APPROVED** [review]
Nice! Looks good to me :)

File: examples/full-jaffle-shop-demo/dbt/models/plan.yml [hunk 1]
```diff
@@ -3,4 +3,22 @@ models:
...
\ No newline at end of file
+      - name: plan_name
+      - name: metadata
+        meta:
+          dimension:
+            hidden: true
+          additional_dimensions:
+            created_by:
+              type: string
+              sql: ${plan.metadata}->>'created_by'
```
Comment (IrakliJani) [inline examples/full-jaffle-shop-demo/dbt/models/plan.yml:14]: cool!

File: examples/full-jaffle-shop-demo/dbt/models/plan.yml [hunk 2]
```diff
@@ -3,4 +3,22 @@ models:
   - name: plan
     columns:
       - name: id
-      - name: plan_name
\ No newline at end of file
+      - name: plan_name
+      - name: metadata
+        meta:
+          dimension:
+            hidden: true
```
Comment (IrakliJani) [inline examples/full-jaffle-shop-demo/dbt/models/plan.yml:10]: what if this dimension has custom sql? will that still work?

also.. I can't think of a different type than json and arrays for base dimension. is there another usecase for this?

File: packages/common/src/compiler/translator.ts [hunk 3]
```diff
@@ -87,7 +87,11 @@ const convertDimension = (
     source?: Source,
     timeInterval?: TimeFrames,
     startOfWeek?: WeekDay | null,
+    isAdditionalDimension?: boolean,
 ): Dimension => {
+    if (isAdditionalDimension) {
+        console.log('additional ', column.name);
+    }
```
Comment (ZeRego) [inline packages/common/src/compiler/translator.ts:94]:
```suggestion

```

File: examples/full-jaffle-shop-demo/dbt/models/plan.yml (hunk 2 above)
Comment (ZeRego) [reply examples/full-jaffle-shop-demo/dbt/models/plan.yml:10]: > what if this dimension has custom sql? will that still work?

it still works

> is there another usecase for this?

There are a few use cases. You can see more in the docs https://docs.lightdash.com/references/dimensions#additional-dimensions

**Pull request merged**